internal/gps/solver.go

```diff
@@ -987,11 +985,8 @@ func (s *solver) getLockVersionIfValid(id ProjectIdentifier) (Version, error) {
 			//}
 		}
 
-		if !found {
-			// No match found, which means we're going to be breaking the lock
-			s.b.breakLock()
```

Comment:
We still want this call, as this is what triggers prefetching of everything that's in the root lock, but hasn't already been selected. All that really needs dropping is the `return nil, nil`.